waveorder/focus.py

```diff
@@ -0,0 +1,104 @@
+from waveorder import util
+import numpy as np
+
+
+def focus_from_transverse_band(
+    zyx_array,
+    NA_det,
+    lambda_ill,
+    pixel_size,
+    midband_fractions=(0.125, 0.25),
+    mode="max",
```

Comment:
Not critical, but could try:
```py
from typing import Literal

def focus(mode: Literal["min", "max"] = "max")
```

Reply:
In vanilla Python function signatures a type hint is just a 'hint'. (But can at least help with static analysis and auto-populate the doc-string if you hinted your IDE.) Checks are not enforced:
```pycon
In [1]: from typing import Literal

In [2]: def lit(v: Literal["0.1", "0.4"]):
   ...:     print(v)

In [3]: lit("0.1")
0.1

In [4]: lit("0.2")
0.2
```

But this allows for both static and runtime type checkers (e.g. pydantic) to check the data if we add them in the future.